src/main/java/com/linecorp/armeria/common/ServiceInvocationContext.java

```diff
@@ -43,9 +45,35 @@
 
     /**
      * Returns the context of the invocation that is being handled in the current thread.
+     *
+     * @throws IllegalStateException if the context is unavailable in the current thread
      */
-    public static Optional<ServiceInvocationContext> current() {
-        return Optional.ofNullable(context.get());
+    public static ServiceInvocationContext current() {
+        final ServiceInvocationContext ctx = context.get();
+        if (ctx == null) {
+            throw new IllegalStateException(ServiceInvocationContext.class.getSimpleName() + " unavailable");
+        }
+        return ctx;
+    }
+
+    /**
+     * Maps the context of the invocation that is being handled in the current thread.
+     *
+     * @param mapper the {@link Function} that maps the invocation
+     * @param fallback the {@link Supplier} that provides the value when the context is unavailable
+     *                 in the current thread. If {@code null}, {@code () -> null} is used instead.
```

Comment:
I guess these would be the docs, but they're long.

http://types.cs.washington.edu/checker-framework/current/checker-framework-manual.html#nullness-checker

In general, I think a project determines what their default nullness is and adds an annotation when breaking from it. This project (as all good projects should) should have default non-null, meaning we add @Nullable when there is something that may be null. This means method parameters, return types, and fields. Basically anything which may have been Optional instead.

I guess we should look into adding this to our mvn test, hopefully we've been diligent enough that it can be applied without too many annotation additions.
http://types.cs.washington.edu/checker-framework/current/checker-framework-manual.html#maven

And my personal opinion is there isn't a big difference between Optional and @Nullable, except that Optional may allocate more objects and add some obtuse syntax, so I'd be totally happy with using more nullables instead in general ;)